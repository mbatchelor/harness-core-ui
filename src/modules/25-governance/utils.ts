import type { GovernanceMetadata } from 'services/cd-ng'

interface DoesGovernanceMetaDataHasErrorOrWarningRtn {
  governanceMetaDataHasError: boolean
  governanceMetaDataHasWarning: boolean
}
export const doesGovernanceHasErrorOrWarning = (
  governanceMetadata: GovernanceMetadata | undefined
): DoesGovernanceMetaDataHasErrorOrWarningRtn => {
  let governanceMetaDataHasWarning = false
  let governanceMetaDataHasError = false
  if (governanceMetadata && governanceMetadata?.status === 'error') {
    governanceMetaDataHasError = true
  }
  if (governanceMetadata && governanceMetadata?.status === 'warning') {
    governanceMetaDataHasWarning = true
  }
  return { governanceMetaDataHasError, governanceMetaDataHasWarning }
}
