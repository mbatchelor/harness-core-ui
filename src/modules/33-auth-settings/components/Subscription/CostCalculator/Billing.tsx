/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState } from 'react'
import { Text, Layout, PillToggle } from '@harness/uicore'
import { FontVariation, Color } from '@harness/design-system'
import { capitalize } from 'lodash-es'
import { useStrings } from 'framework/strings'
import type { Module } from 'framework/types/ModuleName'
import { TIME_TYPE } from '@auth-settings/pages/subscriptions/plans/planUtils'
import { FFEquation } from './FFEquation'
import { PremiumSupport } from './PremiumSupport'
import css from './CostCalculator.module.scss'

const Header = ({ time, setTime }: { time: TIME_TYPE; setTime: (time: TIME_TYPE) => void }): React.ReactElement => {
  const { getString } = useStrings()
  return (
    <Layout.Vertical className={css.billingHeader} padding={{ top: 'large', bottom: 'large' }}>
      <Text padding={{ bottom: 'large' }} font={{ variation: FontVariation.BODY1, weight: 'bold' }}>
        {getString('common.subscriptions.tabs.billing')}
      </Text>
      <Layout.Horizontal flex={{ alignItems: 'center', justifyContent: 'start' }}>
        <PillToggle
          onChange={timeClicked => setTime(timeClicked)}
          options={[
            { label: capitalize(TIME_TYPE.YEARLY), value: TIME_TYPE.YEARLY },
            {
              label: capitalize(TIME_TYPE.MONTHLY),
              value: TIME_TYPE.MONTHLY
            }
          ]}
          selectedView={time}
        />
        {time === TIME_TYPE.MONTHLY && (
          <Text
            icon="dollar"
            iconProps={{ color: Color.GREEN_700, size: 12 }}
            font={{ variation: FontVariation.SMALL }}
            padding={{ left: 'small' }}
          >
            {getString('authSettings.costCalculator.yearlySave')}
          </Text>
        )}
      </Layout.Horizontal>
    </Layout.Vertical>
  )
}

function getEquationByModule(module: Module): React.ReactElement {
  switch (module) {
    case 'cf': {
      return <FFEquation />
    }
    default:
      return <></>
  }
}

export const Billing = ({ module, initialTime }: { module: Module; initialTime: TIME_TYPE }): React.ReactElement => {
  const [time, setTime] = useState<TIME_TYPE>(initialTime)
  const [prem, setPrem] = useState<boolean>(false)
  const premDisabled = time === TIME_TYPE.MONTHLY
  return (
    <Layout.Vertical spacing={'large'} padding={{ bottom: 'large' }}>
      <Header
        time={time}
        setTime={(newTime: TIME_TYPE) => {
          setTime(newTime)
          if (newTime === TIME_TYPE.MONTHLY) {
            setPrem(false)
          }
        }}
      />
      {getEquationByModule(module)}
      <PremiumSupport disabled={premDisabled} onChange={setPrem} value={prem} time={time} setTime={setTime} />
    </Layout.Vertical>
  )
}
