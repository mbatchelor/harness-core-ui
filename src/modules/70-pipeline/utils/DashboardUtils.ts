/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { TooltipFormatterContextObject } from 'highcharts'

interface PointStats {
  deployments?: {
    failure?: number
    success?: number
    total?: number
  }
  time?: number
}

/* istanbul ignore next */
export const renderTooltipContent = ({
  time,
  failureRate,
  count,
  successCount,
  failureCount
}: {
  time: string | number
  failureRate: string | number
  count?: number
  successCount?: number
  failureCount?: number
}) => {
  return `<div style="padding: 16px; color: white; width: 282px; height: 128px;">
        <div style="display: flex; justify-content: space-between; border-bottom: 0.5px solid rgba(243, 243, 250); padding-bottom: 7px; margin-bottom: 15px;">
          <div style="font-weight: normal; font-size: 12px; line-height: 18px; opacity: 0.8;">${time}</div>
          <div>
            <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Deployments: </span>
            <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${count}</span>
          </div>
        </div>
        <div style="display: flex; justify-content: space-between;">
          <div>
            <p style="font-weight: 500; font-size: 10px; line-height: 14px; letter-spacing: 0.2px; color: #D9DAE6; margin-bottom: 0px;">Failure Rate</p>
            <p style="font-weight: 600; font-size: 28px; line-height: 38px; color: #FBE6E4;">${failureRate}</p>
          </div>
          <div style="margin-right: 8px;">
            <div style="display: flex; align-items: center; margin-bottom: 6px;">
              <div style="height: 6px; width: 12px; background-color: #5FB34E; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px; opacity: 0.8;">Success </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px;">${successCount}</span>
            </div>
            <div style="display: flex; align-items: center;">
              <div style="height: 6px; width: 12px; background-color: #EE5F54; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Failed </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${failureCount}</span>
            </div>
          </div>
        </div>
      </div>`
}

/* istanbul ignore next */
export const getTooltip = (currPoint: TooltipFormatterContextObject): string => {
  const custom = currPoint?.series?.userOptions?.custom
  const point: PointStats = custom?.[currPoint.key]
  const time =
    point && point?.time
      ? new Date(point?.time).toLocaleDateString('en-US', { day: 'numeric', month: 'long' })
      : currPoint.x
  let failureRate: string | number = 'Infinity'
  if (point?.deployments?.failure && point.deployments?.total) {
    failureRate = ((point.deployments.failure / point.deployments.total) * 100).toFixed(1) + '%'
  }
  if (point?.deployments?.failure === 0) {
    failureRate = '0'
  }
  return renderTooltipContent({
    time,
    failureRate,
    count: point?.deployments?.total,
    successCount: point?.deployments?.success,
    failureCount: point?.deployments?.failure
  })
}
