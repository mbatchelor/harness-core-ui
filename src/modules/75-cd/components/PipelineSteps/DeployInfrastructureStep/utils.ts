/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructure
  onUpdate?: (data: PipelineInfrastructure) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructure
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructure>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructure): boolean {
  if (getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME && !isEmpty(data.environmentRef)) {
    return true
  } else if (data.environment && !isEmpty(data.environment.identifier)) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
