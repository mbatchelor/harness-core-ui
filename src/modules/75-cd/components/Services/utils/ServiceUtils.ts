/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { set } from 'lodash-es'
import type { PipelineInfoConfig, ServiceDefinition } from 'services/cd-ng'

export enum ServiceTabs {
  SUMMARY = 'summaryTab',
  Configuration = 'configuration',
  REFERENCED_BY = 'refrencedByTab',
  ActivityLog = 'activityLog'
}

export const DefaultNewStageName = 'Stage Name'
export const DefaultNewStageId = 'stage_id'
export const DefaultNewServiceId = '-1'

export const newServiceState = {
  service: {
    name: '',
    identifier: '',
    description: '',
    gitOpsEnabled: false,
    tags: {},
    serviceDefinition: {
      type: '' as ServiceDefinition['type'],
      spec: {}
    }
  }
}

const DefaultService = {
  serviceDefinition: {
    spec: {}
  },
  gitOpsEnabled: false
}

export const initialServiceState = {
  service: { ...DefaultService }
  // isLoading: false,
  // isBETemplateUpdated: false,
  // isDBInitialized: false,
  // isUpdated: false,
  // isInitialized: false,
  // gitDetails: {},
  // entityValidityDetails: {}
}

export const setNameIDDescription = (draftData: PipelineInfoConfig, updatedData: PipelineInfoConfig): void => {
  set(draftData, 'identifier', updatedData.identifier)
  set(draftData, 'name', updatedData.name)
  set(draftData, 'description', updatedData.description)
  set(draftData, 'tags', updatedData.tags)
}
