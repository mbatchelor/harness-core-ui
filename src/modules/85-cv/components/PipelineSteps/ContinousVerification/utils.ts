/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, RUNTIME_INPUT_VALUE, SelectOption } from '@wings-software/uicore'
import { isEmpty, isNull, isUndefined, omit, omitBy, set } from 'lodash-es'
import { FormikErrors, yupToFormErrors } from 'formik'
import * as Yup from 'yup'
import type { UseStringsReturn } from 'framework/strings'
import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
import type { StringsMap } from 'stringTypes'
import type { ContinousVerificationData, spec, VerifyStepMonitoredService } from './types'
import {
  VerificationSensitivityOptions,
  durationOptions,
  baseLineOptions,
  trafficSplitPercentageOptions,
  SensitivityTypes,
  defaultMonitoredServiceSpec,
  monitoredServiceRefPath
} from './constants'
import { MONITORED_SERVICE_TYPE } from './components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/SelectMonitoredServiceType/SelectMonitoredServiceType.constants'

/**
 * checks if a field is a runtime input.
 * @param field
 * @returns boolean
 */
export function checkIfRunTimeInput(field: string | SelectOption | number | undefined): boolean {
  return getMultiTypeFromValue(field as string) === MultiTypeInputType.RUNTIME
}

/**
 * validates a particular field for errors.
 * @param fieldValue
 * @param fieldKey
 * @param data
 * @param errors
 * @param getString
 * @param isRequired
 */
export function validateField(
  fieldValue: string,
  fieldKey: string,
  data: ContinousVerificationData,
  errors: any,
  getString: UseStringsReturn['getString'],
  isRequired = true
): void {
  if (checkIfRunTimeInput(fieldValue) && isRequired && isEmpty(data?.spec?.spec && data?.spec?.spec[fieldKey])) {
    set(errors, `spec.spec.${fieldKey}`, getString('fieldRequired', { field: fieldKey }))
  }
}

export function validateMonitoredService(
  data: ContinousVerificationData,
  errors: FormikErrors<ContinousVerificationData>,
  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string,
  isRequired: boolean,
  monitoredService?: VerifyStepMonitoredService
): void {
  if (
    checkIfRunTimeInput(monitoredService?.spec?.monitoredServiceRef) &&
    isRequired &&
    isEmpty(data?.spec?.monitoredService?.spec?.monitoredServiceRef)
  ) {
    set(errors, monitoredServiceRefPath, getString('fieldRequired', { field: 'Monitored service' }))
  }
}

/**
 * validates timeout field.
 * @param template
 * @param getString
 * @param data
 * @param errors
 */
export function validateTimeout(
  getString: UseStringsReturn['getString'],
  data: ContinousVerificationData,
  errors: any,
  template?: ContinousVerificationData,
  isRequired = true
): void {
  if (checkIfRunTimeInput(template?.timeout)) {
    let timeout: Yup.ObjectSchema
    if (isRequired) {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' }).required(getString?.('validation.timeout10SecMinimum'))
      })
    } else {
      timeout = Yup.object().shape({
        timeout: getDurationValidationSchema({ minimum: '10s' })
      })
    }

    try {
      timeout.validateSync(data)
    } catch (e) {
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)
        Object.assign(errors, err)
      }
    }
  }
}

/**
 * returns yaml data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecYamlData(specInfo?: spec, type?: string): spec {
  let validspec = omitBy(specInfo, v => isUndefined(v) || isNull(v) || v === '')
  switch (type) {
    case 'LoadTest':
      validspec = omit(validspec, ['trafficsplit'])
      break
    case 'Bluegreen':
    case 'Canary':
      validspec = omit(validspec, ['baseline'])
      break
    default:
      break
  }

  Object.keys(validspec).map((key: string) => {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (key === 'sensitivity') {
      validspec[key] = validspec[key].value
        ? SensitivityTypes[validspec[key].value as keyof typeof SensitivityTypes]
        : validspec[key]
    } else {
      validspec[key] = validspec[key].value ? validspec[key].value : validspec[key]
    }
  })

  return validspec
}

export function getMonitoredServiceYamlData(spec: ContinousVerificationData['spec']): VerifyStepMonitoredService {
  let monitoredService: VerifyStepMonitoredService = defaultMonitoredServiceSpec

  switch (spec?.monitoredService?.type) {
    case MONITORED_SERVICE_TYPE.DEFAULT:
      monitoredService = defaultMonitoredServiceSpec
      break
    case MONITORED_SERVICE_TYPE.CONFIGURED:
      monitoredService = {
        type: MONITORED_SERVICE_TYPE.CONFIGURED,
        spec: {
          monitoredServiceRef: getMonitoredServiceRef(spec)
        }
      }
      break
    default:
      monitoredService = defaultMonitoredServiceSpec
  }
  return monitoredService
}

export function getMonitoredServiceRef(spec: ContinousVerificationData['spec']): string {
  return spec?.monitoredService?.spec?.monitoredServiceRef as string
}

/**
 * returns forms data for spec field.
 * @param specInfo
 * @returns spec
 */
export function getSpecFormData(specInfo: spec | undefined): spec {
  const validspec: spec | undefined = { ...specInfo }
  if (specInfo) {
    Object.keys(specInfo).map((key: string) => {
      switch (key) {
        case 'sensitivity':
          setFieldData(validspec, 'sensitivity', VerificationSensitivityOptions)
          break
        case 'duration':
          setFieldData(validspec, 'duration', durationOptions)
          break
        case 'baseline':
          setFieldData(validspec, 'baseline', baseLineOptions)
          break
        case 'trafficsplit':
          setFieldData(validspec, 'trafficsplit', trafficSplitPercentageOptions)
          break
        default:
      }
    })
  }
  return validspec
}

/**
 * sets particular field data
 * @param validspec
 * @param field
 * @param options
 */
export function setFieldData(validspec: spec | undefined, field: string, fieldOptions: SelectOption[]): void {
  //finding the complete option if the field is fixed input
  if (validspec && validspec[field] && validspec[field] !== RUNTIME_INPUT_VALUE) {
    //TODO logic in if block will be removed once backend api is fixed : https://harness.atlassian.net/browse/CVNG-2481
    if (field === 'sensitivity') {
      validspec[field] = fieldOptions.find(
        (el: SelectOption) =>
          SensitivityTypes[el.value as keyof typeof SensitivityTypes] === (validspec && validspec[field])
      )
    } else {
      validspec[field] = fieldOptions.find((el: SelectOption) => el.value === (validspec && validspec[field]))
    }
  }
}
