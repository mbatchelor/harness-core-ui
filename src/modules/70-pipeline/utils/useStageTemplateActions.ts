/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import { parse } from 'yaml'
import produce from 'immer'
import { useCallback } from 'react'
import type { StageElementConfig } from 'services/cd-ng'
import { usePipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import { createTemplate, getStageType } from '@pipeline/utils/templateUtils'
import type { TemplateSummaryResponse } from 'services/template-ng'

interface TemplateActionsReturnType {
  addOrUpdateTemplate: (selectedTemplate?: TemplateSummaryResponse) => Promise<void>
  removeTemplate: () => Promise<void>
}

export function useStageTemplateActions(): TemplateActionsReturnType {
  const {
    state: {
      selectionState: { selectedStageId = '' },
      templateTypes
    },
    updateStage,
    getStageFromPipeline,
    getTemplate
  } = usePipelineContext()
  const { stage } = getStageFromPipeline(selectedStageId)

  const addOrUpdateTemplate = useCallback(
    async (selectedTemplate?: TemplateSummaryResponse) => {
      try {
        const { template, isCopied } = await getTemplate({
          templateType: 'Stage',
          selectedChildType: getStageType(stage?.stage, templateTypes),
          selectedTemplate
        })
        const node = stage?.stage
        const processNode = isCopied
          ? produce(defaultTo(parse(template?.yaml || '')?.template.spec, {}) as StageElementConfig, draft => {
              draft.name = defaultTo(node?.name, '')
              draft.identifier = defaultTo(node?.identifier, '')
            })
          : createTemplate(node, template)
        await updateStage(processNode)
      } catch (_) {
        // Do nothing.. user cancelled template selection
      }
    },
    [stage?.stage, templateTypes, getTemplate, updateStage]
  )

  const removeTemplate = useCallback(async () => {
    const node = stage?.stage
    const processNode = produce({} as StageElementConfig, draft => {
      draft.name = defaultTo(node?.name, '')
      draft.identifier = defaultTo(node?.identifier, '')
      draft.type = getStageType(node, templateTypes)
    })
    await updateStage(processNode)
  }, [stage?.stage, templateTypes, updateStage])

  return { addOrUpdateTemplate, removeTemplate }
}
