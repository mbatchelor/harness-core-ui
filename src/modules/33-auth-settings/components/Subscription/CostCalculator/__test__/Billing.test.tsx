/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { TIME_TYPE } from '@auth-settings/pages/subscriptions/plans/planUtils'
import { Billing } from '../Billing'

describe('Billing', () => {
  test('Billing', async () => {
    const { container, getByText, queryByText } = render(
      <TestWrapper>
        <Billing module="cf" initialTime={TIME_TYPE.YEARLY} />
      </TestWrapper>
    )
    expect(queryByText('authSettings.costCalculator.yearlySave')).not.toBeInTheDocument()
    fireEvent.click(getByText('Monthly'))
    await waitFor(() => {
      expect(getByText('authSettings.costCalculator.yearlySave')).toBeInTheDocument()
    })
    expect(container).toMatchSnapshot()
  })
})
