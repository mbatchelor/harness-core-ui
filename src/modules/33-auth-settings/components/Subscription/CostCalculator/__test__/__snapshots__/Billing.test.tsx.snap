// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing Billing 1`] = `
<div>
  <div
    class="Layout--vertical Layout--layout-spacing-large StyledProps--main StyledProps--padding-bottom-large"
  >
    <div
      class="Layout--vertical StyledProps--main billingHeader StyledProps--padding-top-large StyledProps--padding-bottom-large"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--padding-bottom-large StyledProps--font-variation-body1 StyledProps--font-weight-bold"
      >
        common.subscriptions.tabs.billing
      </p>
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-start"
      >
        <div
          class="PillToggle--optionBtns"
        >
          <div
            class="PillToggle--item"
            data-name="toggle-option-one"
            role="button"
            tabindex="0"
          >
            Yearly
          </div>
          <div
            class="PillToggle--item PillToggle--selected"
            data-name="toggle-option-two"
            role="button"
            tabindex="0"
          >
            Monthly
          </div>
        </div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--padding-left-small"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon bp3-icon-dollar StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-green700"
            icon="dollar"
          >
            <svg
              data-icon="dollar"
              height="12"
              viewBox="0 0 16 16"
              width="12"
            >
              <desc>
                dollar
              </desc>
              <path
                d="M12.83 9.51c-.1-.3-.25-.58-.45-.84s-.45-.49-.75-.7c-.3-.2-.65-.36-1.05-.48-.16-.04-.43-.11-.8-.2-.35-.09-.73-.18-1.12-.28-.39-.1-.74-.19-1.06-.27-.31-.08-.49-.12-.54-.13-.43-.12-.78-.29-1.05-.52-.27-.23-.4-.55-.4-.95 0-.29.07-.53.21-.72.14-.19.32-.34.54-.46.22-.11.46-.19.72-.24.26-.05.52-.08.77-.08.74 0 1.35.15 1.83.46.48.3.75.83.81 1.56h2.14c0-.6-.13-1.13-.38-1.58-.25-.45-.59-.84-1.02-1.15-.43-.31-.93-.54-1.49-.7-.24-.06-.49-.1-.75-.14V1c0-.55-.45-1-1-1s-1 .45-1 1v1.08c-.23.03-.46.07-.68.13-.54.13-1.02.34-1.44.61-.42.28-.76.63-1.02 1.05-.26.43-.39.93-.39 1.5 0 .3.04.59.13.88.09.29.23.56.44.82.21.26.48.49.83.7.35.21.79.38 1.31.51.85.21 1.56.38 2.14.52.58.13 1.08.28 1.52.42.25.09.48.23.69.44.21.21.32.53.32.97 0 .21-.05.42-.14.63-.09.21-.24.39-.45.55-.21.16-.47.29-.81.39-.33.1-.73.15-1.2.15-.43 0-.84-.05-1.21-.14-.37-.09-.7-.24-.99-.43-.29-.2-.51-.45-.67-.76-.16-.31-.24-.68-.24-1.12H3c.01.71.15 1.32.43 1.84.27.52.64.94 1.1 1.27.46.33.99.58 1.61.74.27.07.56.12.85.16V15c0 .55.45 1 1 1s1-.45 1-1v-1.05c.3-.03.61-.08.9-.15.58-.13 1.1-.34 1.56-.63.46-.29.83-.66 1.11-1.11.28-.45.42-1 .42-1.64 0-.31-.05-.61-.15-.91z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          authSettings.costCalculator.yearlySave
        </p>
      </div>
    </div>
    <div
      class="Layout--horizontal StyledProps--main equation StyledProps--flex StyledProps--flex-alignItems-start StyledProps--flex-justifyContent-left"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-right-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          authSettings.costCalculator.developerLicenses common.subtotal
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium"
        >
          common.perMonth
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.billedAnnually
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.unitPrice common.perMonth
        </p>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--padding-right-large StyledProps--padding-top-medium StyledProps--font-size-large"
      >
        +
      </p>
      <div
        class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-right-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          authSettings.costCalculator.maus common.subtotal
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium"
        >
          common.perMonth
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.billedAnnually
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.unitPrice common.perMonth
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.firstIncludedFree
        </p>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--padding-right-large StyledProps--padding-top-medium StyledProps--font-size-large"
      >
        =
      </p>
      <div
        class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          authSettings.yearlySubscriptionTotal
        </p>
        <div
          class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start StyledProps--flex-alignItems-baseline"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-medium"
          >
            common.perMonth
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
          >
            authSettings.plusTax
          </p>
        </div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.billedAnnually
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
        >
          authSettings.autoRenewal
        </p>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main premSupport"
    >
      <div
        class="Layout--horizontal StyledProps--main"
      >
        <span
          class="bp3-popover-wrapper"
        >
          <span
            class="bp3-popover-target"
          >
            <label
              class="bp3-control bp3-checkbox bp3-disabled StyledProps--font Checkbox--checkbox checkbox StyledProps--main"
            >
              <input
                disabled=""
                size="12"
                tabindex="0"
                type="checkbox"
              />
              <span
                class="bp3-control-indicator"
              />
            </label>
          </span>
        </span>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-start StyledProps--padding-top-xsmall"
          style="width: 90%;"
        >
          <div
            class="Layout--vertical StyledProps--main"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-start"
            >
              <p
                class="StyledProps--font StyledProps--main crownLabel StyledProps--padding-bottom-xsmall StyledProps--font-variation-small"
                style="display: flex; align-items: center;"
              >
                <span
                  class="bp3-icon bp3-icon-crown StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-orange700"
                  icon="crown"
                >
                  <svg
                    data-icon="crown"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      crown
                    </desc>
                    <path
                      d="M2 6l3 2 3-4 3 4 3-2-1 6H3L2 6zm6-5a1 1 0 110 2 1 1 0 010-2zM1 3a1 1 0 110 2 1 1 0 010-2zm14 0a1 1 0 110 2 1 1 0 010-2zM3 13h10v2H3v-2z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                authSettings.costCalculator.premSupport.title
              </p>
              <p
                class="StyledProps--font StyledProps--main switchToYearly StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-primary6"
                style="display: flex; align-items: center;"
              >
                authSettings.costCalculator.switchToYearly
                <span
                  class="bp3-icon bp3-icon-arrow-right StyledProps--main Text--icon StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-primary6"
                  icon="arrow-right"
                >
                  <svg
                    data-icon="arrow-right"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      arrow-right
                    </desc>
                    <path
                      d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </p>
            </div>
            <ul
              class="premLabel"
            >
              <li>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
                >
                  authSettings.costCalculator.premSupport.line1
                </p>
              </li>
              <li>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
                >
                  authSettings.costCalculator.premSupport.line2
                </p>
              </li>
              <li>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
                >
                  authSettings.costCalculator.premSupport.line3
                </p>
              </li>
            </ul>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall"
          >
            authSettings.costCalculator.premSupport.includedByDefault
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
